### DISCUSSION TITLE
example for using qtree with a qbreadcrump

### DISCUSSION THREAD

**Discussion by donnyJ42** [quasarframework/quasar#9273] [General - Components / Directives / etc]
Hello,

I like to use the qTree as navigation and need a breadcrumb for this and would like to use the qbreadcrumbs.
Is there an example how can I combine the 2 quasar components for a breadcrumb to navigate inside the tree?
My qTree structure, wich comes from a json, could looks like the following, I dont know how much parents or childs I get
and the breadcrump should looks like:

parent 1 -> child 1.1 -> child 1.2 -> childofchild 1.2.1 
 
All examples I found about breadcrumps are static with 3-4 links or just with breadcrump data from meta fields in routes.js

I think I need an array of the selected ids of the nodes as paths and then use them in my qBreadcrumb, so I started to play with:

https://codepen.io/donnyj42/pen/wvJGPQZ

But if I expand the 2. parent and all its childnodes and then close the 2. parent, the child (2.3) will stay in the array of keys of expanded nodes? Can I close all childnodes if the parent node is closed?

And another question I have is, how to build the links for the breadcrump with this data, I know building links from route.params but only for simple linking not if I dont know how much links I have.

Thanks in advance

**Comment by TobyMosque** [comment]
how u breadcrumb will look like if two or more branchs of the Qtree are expanded?

**Reply by donnyJ42** [reply]
Thats why I ask.

**Comment by hawkeye64** [comment]
![image](https://user-images.githubusercontent.com/10262924/118318249-5a7b2700-b4b6-11eb-8fdb-44009b40b9e6.png)
It doesn't look like `child 2.2` is a child of `child 2.1`, so not sure if your example would work. Maybe a typo. If so, you can look at code I wrote for this repo, which has what I think you are asking for: https://github.com/hawkeye64/electron-quasar-file-explorer

**Reply by donnyJ42** [reply]
@hawkeye64 thats a really nice advanced example thanks for your work and sharing the code here, I will try to get it running and to extract the breadcrump. Seems not compatible with win64, I tryed to build the app but get the error: `info fsevents@2.3.2: The platform "win32" is incompatible with this module.` if I run quasar dev

**Reply by hawkeye64** [reply]
@donnyJ42 fsevents is a package for Macs only. Everyone always sees that warning on Linux and Windows-based systems.

**Reply by donnyJ42** [reply]
Yes I know, but I cant build a clone of your repro to run it in dev mode locally

I just checked the code out and run then `npm i` and then `quasar dev`

![fexplorer_build](https://user-images.githubusercontent.com/84176652/118395512-13726a80-b64b-11eb-9031-bc1dcdd4b775.jpg)
